RTOS/nuttx/wolfssl/README.md

```diff
@@ -0,0 +1,202 @@
+    $ cd ~
+    $ mkdir nuttxspace
+    ```
+2) Install dependencies:
+    ```
+    $ cd ~/nuttxspace
+    $ sudo apt install -y bison flex gettext texinfo libncurses5-dev libncursesw5-dev gperf automake libtool pkg-config build-essential gperf genromfs libgmp-dev libmpc-dev libmpfr-dev libisl-dev binutils-dev libelf-dev libexpat-dev gcc-multilib g++-multilib picocom u-boot-tools util-linux
+    $ sudo apt install -y kconfig-frontends
+    $ sudo apt install -y gcc-arm-none-eabi binutils-arm-none-eabi
+    ```
+3) Clone nuttx and nuttx-apps into working directory:
+    ```
+    $ git clone https://github.com/apache/nuttx.git nuttx
+    $ git clone https://github.com/apache/nuttx-apps apps
+    ```
+4) Copy this directory into the working directory applications:
+    ```
+    $ cp -R RTOS/nuttx/wolfssl ~/nuttxspace/apps/crypto/wolfssl
+    ```
+5) Setup wolfSSL in preparation for the build, `WOLFSSL_DIR` must be the path to the original wolfssl repo:
+    ```
+    $ cd ~/nuttxspace/apps/crypto/wolfssl
+    $ WOLFSSL_DIR=<path-to-wolfssl-repo> ./setup-wolfssl.sh
+    ```
+6) Setup baseline NuttX configuration (board + NuttX Shell):
+    ```
+    $ cd ~/nuttxspace/nuttx
+    $ ./tools/configure.sh -l <board>:nsh
+    ```
+   If you are using wolfSSL for TLS you should use the `netnsh` target if your board supports it
+   ```
+    $ ./tools/configure.sh -l <board>:netnsh
+   ```
+> **EXAMPLES:**
+>   - For NuttX Simulator: `$ ./tools/configure.sh sim:nsh`
+>   - For BL602 (RISC-V): `$ ./tools/configure.sh -l bl602evb:nsh`
+>   - For NUCLEO-L552ZE-Q (Cortex-M33): `$ ./tools/configure.sh -l nucleo-l552ze:nsh`
+>   - For NUCLEO-H753ZI: `$ ./tools/configure.sh -l nucleo-h743zi:nsh`
+>   - For NUCLEO-F756ZG: `./tools/configure.sh -l nucleo-144:f746-nsh`
+
+7) Start custom configuration system:
+    ```
+    $ make menuconfig
+    ```
+8) Configure NuttX to enable the wolfSSL crypto library test applications:
+    - From main menu select: **Application Configuration > Cryptography Library Support**
+    - Enable and then select **wolfSSL SSL/TLS Cryptography Library**
+    - Enable and then select **wolfSSL applications**
+    - Enable applications:
+        - **wolfCrypt Benchmark application**
+        - **wolfCrypt Test application**
+        - **wolfSSL client and server example**
+    - Select Save from bottom menu, saving to `.config` file
+    - Exit configuration tool
+
+    If you are using wolfSSL for TLS you should use the `netnsh` target and should enable an NTP or some for of system time keeping so that wolfSSL has the current date to check certificates. You will also need to set the right networking settings for NuttX to connect to the internet.
+9) Build NuttX and wolfSSL:
+    ```
+    $ make
+    ```
+    Example Output:
+    ```
+CC:  wolfssl/wolfcrypt/src/random.c /home/john/Documents/nuttxspace/apps/crypto/wolfssl/wolfssl/wolfcrypt/src/random.c:3781:10: warning: #warning "write a real random seed!!!!, just for testing now" [-Wcpp]
```

Comment:
This example make output is very poor. Either fix the warnings or don't include it...